### ISSUE TITLE
Compiler hangs during compilation

### ISSUE THREAD

**Issue by chrislailee** [lampepfl/dotty#12878]
Refer to second issue described in [Issue 12876](https://github.com/lampepfl/dotty/issues/12876).

When compile under sbt, it reports a number of errors. The last of which is the following. Then the console hangs at compile incremental with the timer keeps ticking:

```scala
[error] 457 | } yield passwordOk
[error] | ^
[error] |Could not find an instance of FlatMap for cats.Id.
[error] |I found:
[error] |
[error] | cats.Invariant.catsStdCommutativeMonadForTuple11[A0, A1, A2, A3, A4, A5, A6, A7
[error] | ,
[error] | A8, A9]
[error] |
[error] |But method catsStdCommutativeMonadForTuple11 in trait NTupleMonadInstances1 does not match type cats.FlatMap[cats.Id].
[error] -- [E008] Not Found Error: xxx.scala:48:25
[error] 48 | import scala.reflect.runtime.universe
[error] | ^^^^^^^^^^^^^^^^^^^^^
[error] | value runtime is not a member of reflect

| => root / Compile / compileIncremental 457s
```

Please refer to Issue 12876 for stack traces of the compiler process when hang occurs.

**Comment by chrislailee** [comment]
The problem can be reproduced by compiling project at: https://github.com/chrislailee/CatsTest3.git.

**Comment by smarter** [comment]
Thanks for the reproduction. Interestingly I can reproduce the problem on Java 11 and 16 but not 8. I'm not sure what the issue was, however it appears to be fixed in 3.0.1-RC1 where I can't reproduce it at all, so I'll close this.

**Comment by chrislailee** [comment]
Thanks smarter for looking into this. Great to hear that new version already fixed this.
